src/main/java/org/opensearch/neuralsearch/rest/RestNeuralStatsAction.java

```diff
@@ -190,17 +194,22 @@ private void processStatsRequestParameters(RestRequest request, NeuralStatsInput
         for (String stat : stats) {
             // Validate parameter
             String normalizedStat = stat.toLowerCase(Locale.ROOT);
-            if (isValidParamString(normalizedStat) == false) {
+            if (isValidParamString(normalizedStat) == false || isValidEventOrInfoStatName(normalizedStat) == false) {
                 invalidStatNames.add(normalizedStat);
                 continue;
             }
 
-            if (EVENT_STAT_NAMES.contains(normalizedStat)) {
-                neuralStatsInput.getEventStatNames().add(EventStatName.from(normalizedStat));
-            } else if (STATE_STAT_NAMES.contains(normalizedStat)) {
+            Version minClusterVersion = neuralSearchClusterUtil.getClusterMinVersion();
```

Comment:
Why are we getting this minClusterVersion here? Could you check the code of NeuralQueryBuilder how we use MinClusterUtil class to validate the version?

Reply:
The way we store the min version for stats is slightly different than for other versioned fields. in NeuralQueryBuilder we check MinClusterUtil using a helper that does a map get on the field name. So the field name maps directly to the version. For stats, because we have multiple stats per version, we map the Version to the set of stats available in that version.

I think we can avoid using `getClusterMinVersion()` directly here, changed to refactor it behind MinClusterVersionUtil.

Reply:
`getClusterMinVersion` is better than having static method from `MinClusterVersionUtil` so that we can mock it easily in unit test.